janitor/functions/utils.py

```diff
@@ -653,23 +651,22 @@ def _select(
 
     Returns a DataFrame.
     """
-    assert axis in {"both", "index", "columns"}
-    if axis == "both":
-        if rows is None:
-            rows = slice(None)
-        else:
-            rows = _select_index([rows], df, axis="index")
-        if columns is None:
-            columns = slice(None)
-        else:
-            columns = _select_index([columns], df, axis="columns")
+    if rows is None:
+        rows = slice(None)
+    else:
+        axis = "index"
+        rows = _select_index([rows], df, axis=axis)
+    if columns is None:
+        columns = slice(None)
+    else:
+        axis = "columns"
+        columns = _select_index([columns], df, axis=axis)
+    mapping = {"index": rows, "columns": columns}
+    if not invert:
         return df.iloc[rows, columns]
-    indices = _select_index(list(args), df, axis)
-    if invert:
-        rev = np.ones(getattr(df, axis).size, dtype=np.bool_)
-        rev[indices] = False
-        return df.iloc(axis=axis)[rev]
-    return df.iloc(axis=axis)[indices]
+    rev = np.ones(getattr(df, axis).size, dtype=np.bool_)
+    rev[mapping[axis]] = False
+    return df.iloc(axis=axis)[rev]
```

Comment:
I'm not a big fan of the implementation here (sorry!), I think there's too much hidden assumptions in the function: for example -- that only one of rows or columns can be specified (but not both, and not neither) when invert is True.

Granted this is an internal function, so perhaps it's not a big deal, but I still think it makes maintaining the code in the future unnecessarily harder.

I'ld just build up `rev` + compose the iloc for rows and columns separately before returning the df, to get rid of the aforementioned underlying assumption. WDYT?